cmd/maya-apiserver/spc-watcher/select_disk.go

```diff
@@ -0,0 +1,270 @@
+// so as to provision storagepool
+
+func diskSelector(nodeDiskMap map[string]*nodeDisk, poolType string) []string {
+
+	// selectedDisk will hold a list of disk that will be used to provision storage pool, after a
+	// minimum number of node qualifies
+	var selectedDisk []string
+
+	// requiredDiskCount will hold the required number of disk that should be selcted from a qualified
+	// node for specific pool type
+	var requiredDiskCount int
+
+	// Range over the nodeDiskMap map to get the list of disks
+	for _, val := range nodeDiskMap {
+		// If pool type is striped, 1 disk should be selected
+		if poolType == string(v1alpha1.PoolTypeStripedCPK) {
+			requiredDiskCount = int(v1alpha1.StripedDiskCountCPK)
+		}
+		// If pool type is mirrored, 2 disks should be selected
+		if poolType == string(v1alpha1.PoolTypeMirroredCPK) {
+			requiredDiskCount = int(v1alpha1.MirroredDiskCountCPK)
+			// If the current disk count on the node is less than the required disks
+			// then this is a dirty node and it will not qualify.
+			if len(val.diskList) < requiredDiskCount {
+				continue
+			}
+		}
+		// Select the required disk from qualified nodes.
+		for i := 0; i < requiredDiskCount; i++ {
+			selectedDisk = append(selectedDisk, val.diskList[i])
+		}
+	}
+	return selectedDisk
+}
+
+// diskFilterConstraint will form labels that will be used to filter disks
+
+func diskFilterConstraint(diskType string) string {
+	var label string
+	if diskType == string(v1alpha1.TypeSparseCPK) {
+		label = string(v1alpha1.DiskTypeCPK) + "=" + string(v1alpha1.TypeSparseCPK)
+	} else {
+		label = string(v1alpha1.DiskTypeCPK) + "=" + string(v1alpha1.TypeDiskCPK)
+	}
+	return label
+}
+
+// form usedDisk map that will hold the list of all used disks
+
+func (k *clientSet) getUsedDiskMap() (error, map[string]int) {
+	// Get the list of disk that has been used already for pool provisioning
+	spList, err := k.oecs.OpenebsV1alpha1().StoragePools().List(mach_apis_meta_v1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("unable to get the list of used disks:%v", err), nil
+	}
+	// Form a map that will hold all the used disk
+	usedDiskMap := make(map[string]int)
+	for _, sp := range spList.Items {
+		for _, usedDisk := range sp.Spec.Disks.DiskList {
+			usedDiskMap[usedDisk]++
+		}
+
+	}
+	return nil, usedDiskMap
+}
+
+// form usedNode map to keep a track of nodes on the top of which storagepool cannot be provisioned for a
+// given storagepoolcalim
+
+func (k *clientSet) getUsedNodeMap(spc string) (error, map[string]int) {
+	// Get the list of disk that has been used already for pool provisioning
+	spcSpList, err := k.oecs.OpenebsV1alpha1().StoragePools().List(mach_apis_meta_v1.ListOptions{LabelSelector: string(v1alpha1.StoragePoolClaimCPK) + "=" + spc})
+	if err != nil {
+		return fmt.Errorf("unable to get the list of storagepools for stragepoolclaim %s:%v", spc, err), nil
+	}
+	// Form a map that will hold all the nodes on the top of which pool cannot be provisioned
```

Comment:
Change the wordings to say "on top of which pools have been already provisioned."

Reply:
sure

Reply:
// Form a map that will hold all the nodes where storagepool for the spc has been already created.
How about this?

Reply:
fixed